### ISSUE TITLE
Endpoints: allow user to provide password instead of requiring ssh keys

### ISSUE THREAD

**Issue by atheurer** [perftool-incubator/rickshaw#69]

**Comment by atheurer** [comment]
instead of just assuming a ssh key is installed on the host for the endpoint, allow the use of:
--endpoint <type>,host:<hostname>,password:<password>

**Comment by atheurer** [comment]
This work should start with remotehost endpoint

**Comment by atheurer** [comment]
Using a password will likely require the use of "ssh-askpass"

**Comment by tc-wilson** [comment]
Is this intended to take a plaintext password on the command line?  Since we use keys currently, I imagine many commands within crucible depend on being passwordless right now-- do you want to try and pass that around, or do something more like ssh agent forwarding?

It looks like ssh-askpass is a bit of a beast, but some updates went in at the end of 2020 that make it less picky about DISPLAY and tty (https://bugzilla.mindrot.org/show_bug.cgi?id=69) so I'm hoping we can get v8.4 in all of our supported images or at least build it.

**Comment by atheurer** [comment]
Yes, on the cmdline as a plaintext password, like:
`crucible run uperf --endpoint remotehost,host:this_host,password:this_password,client:1,server:1`

Right now we require/assume an ssh-key is installed and use ssh to run things remotely.

I may be thinking of another utility than ssh-askpass, as this should have nothing to do with X11, let me check.

**Comment by atheurer** [comment]
OK, not ask-sshpass, but **sshpass**

For example:

`sshpass -p <plaintestpassword> ssh -o StrictHostKeyChecking=no perf188 uptime

sshpass only needs to be in the crucible-controller container image.  This is only for the controller to reach endpoint systems (not the other way).
`

**Comment by k-rister** [comment]
FWIW... https://linux.die.net/man/1/sshpass

@atheurer Can you articulate the exact reason we need this?  Is it simply because some people will refuse to use ssh keys for some reason?

**Comment by atheurer** [comment]
If you use ssh keys it is not necessary, but just a convenience if you do not use ssh keys.

**Comment by atheurer** [comment]
This should boil down to:
1) including sshpass in the crucible-controller container image
2) adding a cmdline param to parse password:mypassword for remotehost endpoint script
3) all occurrences of 'ssh' in that remotehost endpoint script needs to conditionally either use what's there now, or use sshpass if the password was provided
4) If we do not already, do a trial ssh in the beginning to ensure the password/ssh_key works

**Comment by atheurer** [comment]
@tc-wilson was this completed?

**Comment by tc-wilson** [comment]
> @tc-wilson was this completed?

No.  There was an issue with building controller images.  I had built one with sshpass that worked locally for me, but when submitting a PR that included having the CI point to my image from quay, CI would not pass.  I think there was an email discussion where I compared the contents of my controller image to the existing controller image that didn't have sshpass  and there were ~10 (non-sshpass) packages that differed and I added those to my image but IIRC there was some larger issue with building controller images that came about from that discussion.  I think I was pointed to bench-tracer work and uperf testing next while the controller image thing was sorted out, and TBH, I haven't kept up with the status of that issue.

**Comment by k-rister** [comment]
Lets continue this after @atheurer and I resolve the current controller rebuild issue.  It seems we sorta dropped the ball on merging the changes last time which probably didn't help.
